R/pagination.R

```diff
@@ -1179,7 +1198,31 @@ paginate_to_mpfs <- function(obj,
     })
   })
 
-  unlist(res, recursive = FALSE)
+  res <- unlist(res, recursive = FALSE)
+
+  # Adding page numbers if needed
+  if (!is.null(page_num)) {
+    total_pages <- length(res)
+    page_str <- gsub("\\{n\\}", total_pages, page_num)
```

Comment:
@clarkliming this is basically your code with some tweaks. Also now it is inside the pagination machinery, so it will be available (if we send down `...`) also in other exports